src/metadata_validator.py

```diff
@@ -350,23 +350,33 @@ def validate_prop_value(self, value, prop_def):
                 if len(errs) > 0:
                     errors.extend(errs)
 
-            elif type ==  "datetime" or type ==  "date":
+            elif type ==  "datetime":
                 try:
-                    val = datetime.strptime(value, '%m/%d/%y %H:%M:%S')
+                    val = datetime.strptime(value, DATETIME_FORMAT)
                 except ValueError as e:
-                    errors.append(create_error("Not a date/datetime", f"The value, {value}, is neither a datetime nor date!"))
+                    errors.append(create_error("Not a valid datetime", f"The value, {value}, is not a valid datetime!"))
+
+            elif type ==  "date":
+                val = None
+                for date_format in DATE_FORMATS:
+                    try:
+                        val = datetime.strptime(value, date_format)
+                        break #if the value can be parsed with the format
+                    except ValueError as e:
+                        continue
+                if val is None:
+                    errors.append(create_error("Not a valid date", f"The value, {value}, is not a valid date!"))
 
             elif type ==  "boolean":
-                try:
-                    val = bool(value)
-                except ValueError as e:
+                if not isinstance(value, bool) and value not in ["yes", "true", "no", "false"]:
                     errors.append(create_error("Not a boolean", f"The value, {value}, is not a boolean!"))
             
             elif type ==  "array":
-                try:
-                    val = list(value)
-                except ValueError as e:
-                    errors.append(create_error("Not a list", f"The value, {value}, is not a list!"))
+                arr = value.split("*")
```

Comment:
Just curious, should we check if this is empty array?

Reply:
Hi Young,  it has been checked if the value is empty before call prop value validation.
  line 225 - 227